### ISSUE TITLE
[Task] Clean definition of Characteristic inheritance

### ISSUE THREAD

**Issue by atextor** [OpenManufacturingPlatform/sds-bamm-aspect-meta-model#148]
**Is your task related to a problem? Please describe.**
Currently there is no clean definition of inheritance for Characteristics, i.e., whether inheritance is allowed, and if so, how it is used in modeling and what it implies. Characteristic inheritance is used implicitly in both the formal specification (c.f. for example the the [definition of bamm-c:Measurement](https://github.com/OpenManufacturingPlatform/sds-bamm-aspect-meta-model/blob/main/src/main/resources/bamm/characteristic/2.0.0/characteristic-definitions.ttl#L151) Characteristic, which includes a line such as `bamm-c:Measurement rdfs:subClassOf bamm-c:Quantifiable`) and in the textual specification (c.f. for example the [Characteristics classes overview diagram](https://openmanufacturingplatform.github.io/sds-bamm-aspect-meta-model/bamm-specification/snapshot/characteristics.html#characteristics-classes)).

**Describe the solution you'd like**

* The currently already used Characteristic inheritance should be made explicit in the textual specification.
* To be consistent with [Entity inheritance](https://openmanufacturingplatform.github.io/sds-bamm-aspect-meta-model/bamm-specification/snapshot/modeling-guidelines.html#extending-entities) and [Property inheritance](https://openmanufacturingplatform.github.io/sds-bamm-aspect-meta-model/bamm-specification/snapshot/modeling-guidelines.html#abstract-entities-with-abstract-properties), the specification should use `bamm:extends` instead of `rdfs:subClassOf`, as this is something that is located on the _model level_, not the _meta model level_.
* The documentation should be extended with modeling examples.

**Comment by georgschmidtdumont** [comment]
Hi @atextor , IMHO we should consider extending the validation as well to ensure that for example `rdfs:subClassOf` is not used.

**Comment by atextor** [comment]
SDS WG discussion 2022-07-28: This issue depends on #160.

**Comment by BirgitBoss** [comment]
Yes, it is irritating, if just reading the documentation, that on the one hand side it looks like inheritance but on the other side dataType is mandatory for Characteristic but optional for Collection that is "inheriting" from Characteristic, Trait, also "inheriting" from "Characteristic" does not have a dataType at all.
